src/lib/swapper/swappers/ZrxSwapper/getZrxTradeQuote/getZrxTradeQuote.ts

```diff
@@ -63,19 +69,34 @@ export async function getZrxTradeQuote<T extends ZrxSupportedChainId>(
       )
     }
 
-    const { average, fast } = await adapter.getGasFeeData()
-
-    // use worst case average eip1559 vs fast legacy
-    const maxGasPrice = bnOrZero(BigNumber.max(average.maxFeePerGas ?? 0, fast.gasPrice))
+    const useSellAmount = !!sellAmount
+    const rate = useSellAmount ? data.price : bn(1).div(data.price).toString()
+    const gasLimit = bnOrZero(data.gas)
 
     // 0x approvals are cheaper than trades, but we don't have dynamic quote data for them.
     // Instead, we use a hardcoded gasLimit estimate in place of the estimatedGas in the 0x quote response.
-    const approvalFeeCryptoBaseUnit = bn(APPROVAL_GAS_LIMIT).times(maxGasPrice).toFixed(0)
+    const { average, approvalFee, txFee } = await (async () => {
+      if (optimism.isOptimismChainAdapter(adapter)) {
+        const { average, l1GasPrice } = await adapter.getGasFeeData()
 
-    const useSellAmount = !!sellAmount
-    const rate = useSellAmount ? data.price : bn(1).div(data.price).toString()
-    const gasLimit = bnOrZero(data.gas)
-    const txFee = gasLimit.times(maxGasPrice)
+        // account for l1 transaction fees for optimism
+        const l1ApprovalFee = bn(OPTIMISM_L1_APPROVE_GAS_LIMIT).times(l1GasPrice)
+        const l1TxFee = bn(OPTIMISM_L1_SWAP_GAS_LIMIT).times(l1GasPrice)
```

Comment:
Add safe l1 gas limit fee buffer to optimism transactions so balance check works and we don't end up with failed transactions down the line